packages/opencensus-exporter-zipkin/src/zipkin.ts

```diff
@@ -138,23 +154,68 @@ export class ZipkinTraceExporter implements Exporter {
    * @param span Span to be translated
    * @param rootSpan Only necessary if the span has rootSpan
    */
-  private translateSpan(span: Span|RootSpan): TranslatedSpan {
+  translateSpan(span: Span|RootSpan): TranslatedSpan {
     const spanTraslated = {
       traceId: span.traceId,
       name: span.name,
       id: span.id,
-      parentId: span.parentSpanId,
-      kind: 'SERVER',
+      // Zipkin API for span kind only accept
+      // (CLIENT|SERVER|PRODUCER|CONSUMER)
+      kind: span.kind === SpanKind.CLIENT ? 'CLIENT' : 'SERVER',
       timestamp: span.startTime.getTime() * 1000,
       duration: Math.round(span.duration * 1000),
       debug: true,
       shared: true,
-      localEndpoint: {serviceName: this.serviceName}
+      localEndpoint: {serviceName: this.serviceName},
+      tags: this.createTags(span.attributes, span.status),
+      annotations: this.createAnnotations(span.annotations, span.messageEvents)
     } as TranslatedSpan;
 
+    if (span.parentSpanId) {
+      spanTraslated.parentId = span.parentSpanId;
+    }
     return spanTraslated;
   }
 
+  /** Converts OpenCensus Attributes ans Status to Zipkin Tags format. */
+  private createTags(
+      attributes: coreTypes.Attributes, status: coreTypes.Status) {
+    const tags: {[key: string]: string} = {};
+    for (const key of Object.keys(attributes)) {
+      tags[key] = String(attributes[key]);
+    }
+    tags[STATUS_CODE] = String(status.code);
+    if (status.message) {
+      tags[STATUS_DESCRIPTION] = status.message;
+    }
+    return tags;
+  }
+
+  /**
+   * Converts OpenCensus Annotation and MessageEvent to Zipkin Annotations
+   * format.
+   */
+  private createAnnotations(
+      annotationTimedEvents: coreTypes.Annotation[],
+      messageEventTimedEvents: coreTypes.MessageEvent[]) {
+    let annotations: Annotation[] = [];
+    if (annotationTimedEvents) {
+      annotations =
+          annotationTimedEvents.map((annotation) => ({
+                                      timestamp: annotation.timestamp * 1000,
```

Comment:
Nit: could you make a constant called `MICROS_PER_MILLI` or similar to use instead of `1000` directly here and the places below to show that the unit conversion is?

Reply:
Done.